sklearn/cross_decomposition/_pls.py

```diff
@@ -442,7 +442,10 @@ def predict(self, X, copy=True):
         X -= self._x_mean
         X /= self._x_std
         Ypred = np.dot(X, self.coef_)
-        return Ypred + self._y_mean
+        Ypred += self._y_mean
+        if Ypred.shape[1] == 1:
```

Comment:
Might the shape of `Ypred` change, if `X` solely contains one observation?

For safety, I would make the following change to be sure accessing `Ypred.shape[1]` won't raise an `IndexError` 
```suggestion
        if Ypred.ndim == 2 and Ypred.shape[1]:
```

Reply:
Hey @jjerphan thanks for pointing that out. I just made a test with X with one observation (`shape = (1, 1)`) and the following value error raises: `ValueError: Found array with 1 sample(s) (shape=(1, 1)) while a minimum of 2 is required.`. Also, just to make sure that this does not happen with `X_test.shape == (1, 1)` I ran the test below and it ran OK. Do you want me to add this test to `test_pls.py`?

```
@pytest.mark.parametrize("Estimator", (PLSRegression, PLSCanonical, CCA))
def test_pls_estimators_single_target_shape_2(Estimator):
    # Check shape for single-target predictions on PLS estimators
    # non-regression test for https://github.com/scikit-learn/scikit-learn/issues/19352
    X = np.random.randn(2, 1)
    Y = np.random.randn(2, 1)

    pls = Estimator(n_components=Y.shape[1])
    pls.fit(X, Y)

    X_test = np.random.randn(1, 1)
    Y_pred = pls.predict(X_test)

    assert Y_pred.shape == (1,)
```

Reply:
Hi @polocorona,

Thanks for your answer. Yes, you can add it to the tests.

Reply:
Following up here with @glemaitre's comment https://github.com/scikit-learn/scikit-learn/pull/20355#issuecomment-885941125.